aioslimproto/server.py

```diff
@@ -21,18 +22,23 @@ class SlimServer:
     def __init__(
         self,
         port: int = 3483,
-        cli_port: Optional[int] = 9090,
-        cli_port_json: Optional[int] = 9091,
+        cli_port: Optional[int] = 0,
+        cli_port_json: Optional[int] = 0,
     ) -> None:
         """
         Initialize SlimServer instance.
 
         control_port: The TCP port for the slimproto communication, default is 3483.
         cli_port: Optionally start a simple Telnet CLI server on this port for compatability
-        with players relying on the server providing this feature. None to disable.
-        cli_port_json: Same as cli port but it's newer JSOn RPC equivalent.
+        with players relying on the server providing this feature. None to disable, 0 for autoselect.
+        cli_port_json: Same as cli port but it's newer JSON RPC equivalent.
         """
         self.logger = logging.getLogger(__name__)
+        # if port is specified as 0, auto select a free port for the cli/json interface
+        if cli_port == 0:
+            cli_port = select_free_port(9090, 9190)
```

Comment:
Side effects on instantiation is not nice. I'd recommend refactoring this so that any side effects are done inside explicit function calls.

Reply:
I will do that in a small follow up improvement. Kind of in a hurry to get some users to test the fixes.
Thanks for your review !